chia/wallet/puzzles/p2_delegated_secp.clsp

```diff
@@ -1,14 +1,15 @@
 ; p2_delegated with SECP256-R1 signature
 ; this is the "standard puzzle" for spending coins with SECP keys (ie secure enclave)
 
-(mod (SECP_PK delegated_puzzle delegated_solution signature)
+(mod (SECP_PK delegated_puzzle delegated_solution signature coin_id genesis_challenge)
   (include *standard-cl-21*)
   (include condition_codes.clib)
   (include sha256tree.clib)
 
-  (if (secp256r1_verify SECP_PK (sha256 (sha256tree delegated_puzzle)) signature)
+  (if (secp256r1_verify SECP_PK (sha256 (sha256tree delegated_puzzle) coin_id genesis_challenge) signature)
```

Comment:
You've got to do `(ASSERT_MY_COIN_ID` or else this can still be replayed in future coins.  Also, you should curry the genesis challenge rather than pass it in as a solution value otherwise it is as pointless as the coin ID without the assertion.

Reply:
Also, to be perfectly aligned with AGG_SIG_ME, you don't need to hash this.  Just use `concat`.

Reply:
Thanks - I had taken out the coin_id assertion when something was failing and neglected to add it back. 
In terms of using `concat`, the function I'm using to generate the signature (`ecdsa.SigningKey.sign_deterministic`) seems to demand a hash of the input data. I'm not sure what the signing process is for the enclave, but can swap it out if there's another lib for this?

Reply:
That's weird, `sign_deterministic` doesn't let you sign a message, only its hash?

Reply:
`sign_deterministic` takes the raw message, and hashes it internally while creating the sig. So then the chialisp secp_verify needs to use the hash of the message to verify against the signature we're passing in.

Reply:
Do we know if this is a limitation of the algorithm?  It can only sign 32 bytes? If so, I'm fine with this, but if it's just the particular python library we're using, then I think we should defer to the cheaper chialisp (no hashing) and figure it out on the next layer up.